roc_jni/src/main/cpp/receiver.cpp

```diff
@@ -73,44 +78,68 @@ JNIEXPORT jlong JNICALL Java_org_rocstreaming_roctoolkit_Receiver_open(JNIEnv *
     return (jlong) receiver;
 }
 
+JNIEXPORT void JNICALL Java_org_rocstreaming_roctoolkit_Receiver_setMulticastGroup(JNIEnv * env, jobject thisObj,
+                    jlong receiverPtr, jint slot, jint interface, jstring jip) {
+    roc_receiver* receiver    = NULL;
+    const char*   ip = NULL;
+
+    receiver = (roc_receiver*) receiverPtr;
+    ip = env->GetStringUTFChars(jip, 0);
+
+    if (roc_receiver_set_multicast_group(receiver, (roc_slot) slot, (roc_interface) interface, ip) != 0) {
+        env->ReleaseStringUTFChars(jip, ip);
+        jclass exceptionClass = env->FindClass(EXCEPTION);
+        env->ThrowNew(exceptionClass, "Couldn't set multicast group");
+        return;
+    }
+    env->ReleaseStringUTFChars(jip, ip);
+}
+
 JNIEXPORT void JNICALL Java_org_rocstreaming_roctoolkit_Receiver_bind(JNIEnv * env, jobject thisObj, jlong receiverPtr,
-                    jint type, jint protocol, jobject jaddress) {
+                    jint slot, jint interface, jobject jendpoint) {
     roc_receiver*   receiver    = NULL;
-    roc_address     address;
+    roc_endpoint*   endpoint    = NULL;
     int             port        = 0;
 
     receiver = (roc_receiver*) receiverPtr;
-
-    if (address_unmarshall(env, &address, jaddress) != 0) {
+    if (endpoint_unmarshal(env, &endpoint, jendpoint) != 0) {
         jclass exceptionClass = env->FindClass(ILLEGAL_ARGUMENTS_EXCEPTION);
         env->ThrowNew(exceptionClass, "Bad arguments");
         return;
     }
 
-    if ((port = roc_address_port(&address)) < 0) {
+    if (roc_endpoint_get_port(endpoint, &port) != 0) {
+        roc_endpoint_deallocate(endpoint);
         jclass exceptionClass = env->FindClass(EXCEPTION);
         env->ThrowNew(exceptionClass, "Error binding receiver");
         return;
     }
 
-    if (roc_receiver_bind(receiver, (roc_port_type) type, (roc_protocol) protocol, &address) != 0) {
+    if (roc_receiver_bind(receiver, (roc_slot) slot, (roc_interface) interface, endpoint) != 0) {
+        roc_endpoint_deallocate(endpoint);
         jclass exceptionClass = env->FindClass(EXCEPTION);
         env->ThrowNew(exceptionClass, "Error binding receiver");
         return;
     }
 
     // ephemeral port
     if (!port) {
```

Comment:
Maybe better `if (port == 0)`? Not sure if port can be `-1` here. But if it can, `!port` will be true.